### PR TITLE
chore: update edge functions ci/cd guide

### PR DESCRIPTION
**Pull request by laktek**
## What kind of change does this PR introduce?

This updates Edge Functions guide to include CI/CD examples for Bitbucket and GitLab CI.

### CODE REVIEW

**Review by sweatybridge — APPROVED** [review]
Lgtm!

File: apps/docs/pages/guides/functions/cicd-workflow.mdx [hunk 1]
```diff
@@ -41,6 +43,68 @@ jobs:
...
+# List of stages for jobs, and their order of execution
+stages:
+  - setup
+  - deploy
+
+# This job runs in the setup stage, which runs first.
+setup-npm:
+  stage: setup
+  script:
+    - npm i supabase@beta
```
Comment (inian) [inline apps/docs/pages/guides/functions/cicd-workflow.mdx:63]: Should we recommend `beta` in the docs @sweatybridge? kinda defeats the point of a staged rollout of the cli if we recommend it by default

Comment (sweatybridge) [reply apps/docs/pages/guides/functions/cicd-workflow.mdx:63]: good catch. lets drop the beta in docs

```suggestion
    - npm i supabase
```

File: apps/docs/pages/guides/functions/cicd-workflow.mdx [hunk 2]
```diff
@@ -41,6 +43,68 @@ jobs:
...
+image: node:18
+
+pipelines:
+  default:
+    - step:
+        name: Setup
+        caches:
+          - node
+        script:
+          - npm i supabase@beta
```
Comment (sweatybridge) [inline apps/docs/pages/guides/functions/cicd-workflow.mdx:97]:
```suggestion
          - npm i supabase
```

**Pull request merged**